sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2SQLSuite.scala

```diff
@@ -1767,6 +1769,213 @@ class DataSourceV2SQLSuite
+           | USING foo
+           | PARTITIONED BY (id, p)
+         """.stripMargin)
+      sql(
+        s"""
+           | INSERT INTO $t
+           | VALUES (1L, 'Herry', 26, 1),
+           | (2L, 'Jack', 31, 2),
+           | (3L, 'Lisa', 28, 3),
+           | (4L, 'Frank', 33, 3)
+         """.stripMargin)
+      sql(s"UPDATE $t SET name='Robert', age=32")
+
+      checkAnswer(spark.table(t),
+        Seq(Row(1, "Robert", 32, 1),
+          Row(2, "Robert", 32, 2),
+          Row(3, "Robert", 32, 3),
+          Row(4, "Robert", 32, 3)))
+    }
+  }
+
+  test("Update: update with where clause") {
+    val t = "testcat.ns1.ns2.tbl"
+    withTable(t) {
+      sql(
+        s"""
+           | CREATE TABLE $t (id bigint, name string, age int, p int)
+           | USING foo
+           | PARTITIONED BY (id, p)
+         """.stripMargin)
+      sql(
+        s"""
+           | INSERT INTO $t
+           | VALUES (1L, 'Herry', 26, 1),
+           | (2L, 'Jack', 31, 2),
+           | (3L, 'Lisa', 28, 3),
+           | (4L, 'Frank', 33, 3)
+         """.stripMargin)
+      sql(s"UPDATE $t SET name='Robert', age=32 where p=2")
+
+      checkAnswer(spark.table(t),
+        Seq(Row(1, "Herry", 26, 1),
+          Row(2, "Robert", 32, 2),
+          Row(3, "Lisa", 28, 3),
+          Row(4, "Frank", 33, 3)))
+    }
+  }
+
+  test("Update: update the partition key") {
+    val t = "testcat.ns1.ns2.tbl"
+    withTable(t) {
+      sql(
+        s"""
+           | CREATE TABLE $t (id bigint, name string, age int, p int)
+           | USING foo
+           | PARTITIONED BY (id, p)
+         """.stripMargin)
+      sql(
+        s"""
+           | INSERT INTO $t
+           | VALUES (1L, 'Herry', 26, 1),
+           | (2L, 'Jack', 31, 2),
+           | (3L, 'Lisa', 28, 3),
+           | (4L, 'Frank', 33, 3)
+         """.stripMargin)
+      sql(s"UPDATE $t SET p=4 where id=4")
+
+      checkAnswer(spark.table(t),
+        Seq(Row(1, "Herry", 26, 1),
+          Row(2, "Jack", 31, 2),
+          Row(3, "Lisa", 28, 3),
+          Row(4, "Frank", 33, 4)))
+    }
+  }
+
+  test("Update: update with aliased target table") {
+    val t = "testcat.ns1.ns2.tbl"
+    withTable(t) {
+      sql(
+        s"""
+           | CREATE TABLE $t (id bigint, name string, age int, p int)
+           | USING foo
+           | PARTITIONED BY (id, p)
+         """.stripMargin)
+      sql(
+        s"""
+           | INSERT INTO $t
+           | VALUES (1L, 'Herry', 26, 1),
+           | (2L, 'Jack', 31, 2),
+           | (3L, 'Lisa', 28, 3),
+           | (4L, 'Frank', 33, 3)
+         """.stripMargin)
+      sql(s"UPDATE $t AS tbl SET tbl.name='Robert', tbl.age=32 where p=2")
+
+      checkAnswer(spark.table(t),
+        Seq(Row(1, "Herry", 26, 1),
+          Row(2, "Robert", 32, 2),
+          Row(3, "Lisa", 28, 3),
+          Row(4, "Frank", 33, 3)))
+    }
+  }
+
+  test("Update: columns aliases is not allowed.") {
```

Comment:
let's test it in the `DDLParserSuite`.

Reply:
Is it suitable move it to `DDLParserSuite`? `Update` is actually a `DML` statement.